hunts/src/EditPuzzleTagsModal.js

```diff
@@ -3,15 +3,23 @@ import { useDispatch, useSelector } from "react-redux";
 import Modal from "react-bootstrap/Modal";
 import Button from "react-bootstrap/Button";
 import Form from "react-bootstrap/Form";
-import { addPuzzleTag } from "./puzzlesSlice";
+import { addPuzzleTag, selectPuzzleTags } from "./puzzlesSlice";
 import { selectHuntTags } from "./huntSlice";
 import { DEFAULT_TAG_COLOR, SELECTABLE_TAG_COLORS } from "./constants";
 import { hideModal } from "./modalSlice";
 import TagPill from "./TagPill";
 import EditableTagList from "./EditableTagList";
 
 function EditPuzzleTagsModal({ puzzleId, puzzleName }) {
-  const allTags = useSelector(selectHuntTags);
+  const huntTags = useSelector(selectHuntTags);
+  const puzzleTags = useSelector(selectPuzzleTags);
+  // Use a map to concat huntTags and puzzleTags and then dedupe.
+  const tagMap = new Map();
+  for (const tag of huntTags.concat(puzzleTags)) {
+    tagMap.set(tag.name, tag);
+  }
+  const allTags = Array.from(tagMap.values());
+
```

Comment:
I think you can dedupe by id instead of name (though since atm tags can't be renamed it's all the same, but I think id is still better)